### PR TITLE
fix: AOC INIT/REVIEW page - ON and OFF time logic.

### LINKED ISSUE #8369: INIT/REVIEW times resetting at touchdown
### Aircraft Version

Development

### Build info

```json
{
    "built": "2024-01-02T11:57:34+00:00",
    "ref": "refs/heads/master",
    "sha": "1f16d82017a1b163a6c4de4db6c7092d6b175ff8",
    "actor": "Benjozork",
    "event_name": "manual",
    "pretty_release_name": "master:1f16d820",
    "version": "v0.12.0-dev.1f16d82"
}
```

### Describe the bug

INIT/REVIEW times resetting at touchdown. After takeoff there are two identical times for OFF and ON fields. ON ground time is equal take OFF time probably because of micro jumps during takeoff roll. IN field is never filled. The whole page resets to empty exactly at touchdown moment. See pictures attached 

### Expected behavior

Some say this page should be reset at next flight. I don't know if it's possible. But it should not resets before full power down or exit a flight.

### Steps to reproduce

1. Do a full flight and look at INIT/REVIEW page

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

Vitold69

### PR DESCRIPTION
**Pull request by rahulrathnakumar**
Partially Fixes #8369

## Summary of Changes

Fixed condition under if (!this.aocTimes.on) to ensure proper triggering of landing block time. 
## Screenshots (if necessary)
Fixed the OFF and ON time discrepancy. 
Microsoft Flight Simulator Screenshot 2024 05 27 - 14 43 18 91

## Additional context

Discord username (if different from GitHub): rahulrk

## Testing instructions

1. Load a flight plan in simbrief
2. Import flight plan into EFB
3. Go to the AOC INIT/REVIEW page, and hit the INIT LSK.
4. Perform the takeoff. Observe the OFF time (Refresh the AOC INIT/REVIEW page by cycling pages).
5. Perform the landing. The ON time field will populate once the aircraft gears make contact. 

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

**Comment by BravoMike99** [comment]
While you are at it, perhaps you might want to fix the block time too? If I'm not mistaken it is populated from simbrief instead of counting the block time (Time elapsed between OUT and IN ).

**Comment by rahulrathnakumar** [comment]
> While you are at it, perhaps you might want to fix the block time too? If I'm not mistaken it is populated from simbrief instead of counting the block time (Time elapsed between OUT and IN ).

Sure thing, I'll fix that too.

**Comment by rahulrathnakumar** [comment]
> > While you are at it, perhaps you might want to fix the block time too? If I'm not mistaken it is populated from simbrief instead of counting the block time (Time elapsed between OUT and IN ).
> 
> Sure thing, I'll fix that too.

I have made the fix for this as well. I think @cptnuss-ops's fix to prevent the page from clearing, combined with my fixes should resolve this issue.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_Init.js [hunk 1]
```diff
@@ -152,8 +152,12 @@ class CDUAocInit {
         if (mcdu.aocTimes.in) {
             inTime = `${FMCMainDisplay.secondsTohhmm(mcdu.aocTimes.in)}[color]green`;
         }
-        if (mcdu.simbrief["blockTime"]) {
-            blockTime = `${FMCMainDisplay.secondsTohhmm(mcdu.simbrief.blockTime)}[color]green`;
+        // Not using simbrief block time anymore
+        // if (mcdu.simbrief["blockTime"]) {
+        //     blockTime = `${FMCMainDisplay.secondsTohhmm(mcdu.simbrief.blockTime)}[color]green`;
+        // }
+        if (mcdu.aocTimes.in && mcdu.aocTimes.out) {
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_Init.js:159]: Please don’t comment out code and leave it. We can easily see the old code in git history if required. Do make sure the PR title accurately reflects what’s in the PR to help people looking at the history in future though.

Comment (rahulrathnakumar) [reply fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/ATSU/A320_Neo_CDU_AOC_Init.js:159]: Removed the old code, thanks for letting me know!

**Review by 2hwk — APPROVED** [review]

**Comment by 2hwk** [comment]
image
Quality Assurance Trainee Report

Discord: 2cas
Object of testing: #8666
Tier of Testing: 1
Date: 14/07/2024

Testing Process: 
- [x] Load a flight plan in simbrief
- [x] Import flight plan into EFB
- [x] Go to the AOC INIT/REVIEW page, and hit the INIT LSK.
- [x] Perform the takeoff. Observe the OFF time (Refresh the AOC INIT/REVIEW page by cycling pages).
- [x] Perform the landing. The ON time field will populate once the aircraft gears make contact.


Negatives:
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Comment by 2hwk** [comment]
@rahulrathnakumar Congratulations on your first merged contribution to the flybywire repository, a staff member should give you the contributor role on the discord shortly.

**Pull request merged**